catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,874 @@
+                            MgmpConstants.CLASSIFICATION_CODE);
+                });
+    }
+
+    private void addImageDescription() {
+
+        Optional<String> cloudCoverage = Optional.ofNullable(metacard.getAttribute(Isr.CLOUD_COVER))
+                .map(Attribute::getValue)
+                .filter(Integer.class::isInstance)
+                .map(Integer.class::cast)
+                .filter(integer -> integer >= MIN_CLOUD_COVERAGE && integer <= MAX_CLOUD_COVERAGE)
+                .map(integer -> Integer.toString(integer));
+
+        boolean isCloudCoverageAvailable = cloudCoverage.isPresent();
+
+        List<Serializable> ratingScaleValues =
+                getSerializables(Isr.NATIONAL_IMAGERY_INTERPRETABILITY_RATING_SCALE);
+
+        boolean isRatingScaleValuesAvailable = ratingScaleValues.size() == 1;
+
+        if (isCloudCoverageAvailable || isRatingScaleValuesAvailable) {
+
+            List<String> attributeDescription = Stream.of(metacard.getAttribute(Isr.COMMENTS))
+                    .flatMap(attribute -> attribute.getValues()
+                            .stream())
+                    .filter(String.class::isInstance)
+                    .map(String.class::cast)
+                    .collect(Collectors.toList());
+
+            if (attributeDescription.size() != 1) {
+                pathValueTracker.add(new Path(MgmpConstants.CONTENT_INFO_MGMP_IMAGE_DESCRIPTION_CLOUD_COVERAGE_ATTRIBUTE_REASON_PATH),
+                        "unknown");
+            } else {
+                pathValueTracker.add(new Path(MgmpConstants.ISR_IMAGE_COMMENT_PATH),
+                        attributeDescription.get(0));
+            }
+
+            pathValueTracker.add(new Path(MgmpConstants.CONTENT_INFO_MGMP_IMAGE_DESCRIPTION_CLOUD_COVERAGE_CONTENT_TYPE_TEXT_PATH),
+                    "Image");
+            pathValueTracker.add(new Path(MgmpConstants.CONTENT_INFO_MGMP_IMAGE_DESCRIPTION_CLOUD_COVERAGE_CONTENT_TYPE_CODE_LIST_PATH),
+                    MgmpConstants.MGMP_COVERAGE_CONTENT_TYPE_CODE);
+            pathValueTracker.add(new Path(MgmpConstants.ISR_IMAGE_DESCRIPTION_PATH), "image");
+        }
+
+        if (!ratingScaleValues.isEmpty()) {
+            ratingScaleValues.stream()
+                    .map(Object::toString)
+                    .findFirst()
+                    .ifPresent(value -> {
+                        pathValueTracker.add(new Path(MgmpConstants.NIIRS_RATING_PATH),
+                                MgmpConstants.NIIRS);
+                        pathValueTracker.add(new Path(MgmpConstants.NIIRS_PATH), value);
+                    });
+
+        }
+
+        addFieldIfString(() -> cloudCoverage, MgmpConstants.CLOUD_COVERAGE_PATH);
+    }
+
+    public void setGmlIdSupplier(Supplier<String> gmlIdSupplier) {
+        this.gmlIdSupplier = gmlIdSupplier;
+    }
+
+    private List<String> replace(List<String> list, String replacement) {
```

Comment:
It's a bit odd to call this `replace()` when you're not really replacing anything.